File: server/src/main/java/by/brawl/entity/NamedEntity.java
package by.brawl.entity;

import javax.persistence.MappedSuperclass;

@MappedSuperclass
public abstract class NamedEntity extends IdEntity {

    protected String name;

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return "NamedEntity{" +
                "name='" + name + '\'' +
                '}';
    }
}

File: server/src/main/java/by/brawl/entity/NamedEntity.kt
package by.brawl.entity

import javax.persistence.MappedSuperclass

@MappedSuperclass
abstract class NamedEntity : IdEntity() {

    lateinit var name: String
        protected set

    override fun toString(): String {
        return "NamedEntity{" +
                "name='" + name + '\'' +
                '}'
    }

}